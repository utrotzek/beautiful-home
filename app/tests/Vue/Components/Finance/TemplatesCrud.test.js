import {mount} from "@vue/test-utils";
import templateCrud from "../../../../resources/js/components/Finance/TemplatesCrud";
import Vue from "vue";
import Vuelidate from "vuelidate";

Vue.use(Vuelidate);

describe("TemplateCrud", () => {
    let wrapper = null;
    const templates = [
        {
            id: 1,
            title: "Standard"
        },
        {
            id: 2,
            title:  "Nicht Standard"
        }
    ];

    beforeEach(() => {
        wrapper = mount(templateCrud,{
            propsData: {
                templates: templates
            }
        });
    });

    it ("renders properly", () => {
        expect(wrapper.element).toMatchSnapshot();
    });

    it("will fire edit event when edit button was clicked", () => {
        wrapper.find(".entry:nth-child(1) .edit-controls .edit").trigger("click");
        expect(wrapper.emitted().editTemplate[0]).toEqual([templates[0]]);
    });

    it("will fire delete event when delete button was clicked", () => {
        wrapper.find(".entry:nth-child(1) .edit-controls .delete").trigger("click");
        expect(wrapper.emitted().deleteTemplate[0]).toEqual([templates[0]]);
    });

    it("will fire create template event when form was submitted properly", () => {
        wrapper.find(".template-crud .template-name").setValue("My new template");
        wrapper.find(".template-crud .submit").trigger("click");
        expect(wrapper.emitted().createTemplate[0]).toEqual(["My new template"]);
    });
});
