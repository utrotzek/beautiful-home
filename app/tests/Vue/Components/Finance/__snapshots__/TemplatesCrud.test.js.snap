// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`TemplateCrud renders properly 1`] = `
<div
  class="template-crud"
>
  <div
    class="row mb-2 entry"
  >
    <div
      class="col-6 title"
    >
      
            Standard
        
    </div>
     
    <div
      class="col-6 edit-controls"
    >
      <button
        class="edit"
        title="Vorlage bearbeiten"
      >
        <i
          class="fa fa-edit"
        />
      </button>
       
      <button
        class="delete"
        title="Vorlage löschen"
      >
        <i
          class="fa fa-trash-alt"
        />
      </button>
    </div>
  </div>
  <div
    class="row mb-2 entry"
  >
    <div
      class="col-6 title"
    >
      
            Nicht Standard
        
    </div>
     
    <div
      class="col-6 edit-controls"
    >
      <button
        class="edit"
        title="Vorlage bearbeiten"
      >
        <i
          class="fa fa-edit"
        />
      </button>
       
      <button
        class="delete"
        title="Vorlage löschen"
      >
        <i
          class="fa fa-trash-alt"
        />
      </button>
    </div>
  </div>
   
  <div
    class="row mt-4"
  >
    <div
      class="col"
    >
      <div
        class="form-inline"
      >
        <div
          class="form-group mx-sm-3 mb-2"
        >
          <label
            class="sr-only"
          >
            Name
          </label>
           
          <input
            class="form-control template-name"
            name="template-name"
            placeholder="Name"
            type="text"
          />
        </div>
         
        <button
          class="btn btn-primary mb-2 submit"
        >
          
                    Vorlage anlegen
                
        </button>
      </div>
    </div>
  </div>
</div>
`;
